### PR TITLE
[DNM] engine: fix reopen failure.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: src/storage/engine/mod.rs [hunk 1]
```diff
@@ -231,6 +233,8 @@ mod tests {
                 .unwrap();
             must_put_cf(e.as_ref(), "cf", b"k", b"v1");
         }
+        // Sleep a while to make sure the `LOCK` file is deleted.
+        thread::sleep(Duration::from_millis(100));
```
Comment (siddontang) [inline src/storage/engine/mod.rs]: using drop can't do it?

Comment (disksing) [inline src/storage/engine/mod.rs]: I also thought it's okay to reopen it as soon as the old one is dropped. But error message shows that the LOCK file is not deleted instantly somehow.

Comment (siddontang) [inline src/storage/engine/mod.rs]: This error occurs in your PR, not in master. 
I guess, In your PR, even the threadpool dropped, the DB may can't be dropped until the thread terminates. 

/cc @BusyJay https://github.com/frewsxcv/rust-threadpool/blob/master/lib.rs#L361
If the channel has un-handled message but the sender is closed, receiver will return error?

**Comment by disksing** [comment]
I have restarted this PR's travis-CI several times, no error showed up again. Seems problem is fixed.
PTAL @siddontang @ngaut

File: src/storage/engine/mod.rs (hunk 1 above)
Comment (disksing) [inline src/storage/engine/mod.rs]: @siddontang My mistake. Thought all PRs were failing because of it... This PR should be closed after I fix #820

Comment (BusyJay) [inline src/storage/engine/mod.rs]: @siddontang No.

Comment (BusyJay) [inline src/storage/engine/mod.rs]: If it happens because of the threadpool, then you can try `Worker`, which allow you shut it down explicitly.

**Pull request closed without merging**